src/inmanta/agent/handler.py

```diff
@@ -495,13 +551,29 @@ def check_facts(self, ctx: HandlerContext, resource: resources.Resource) -> Dict
 
     def available(self, resource: resources.Resource) -> bool:
         """
-        Returns true if this handler is available for the given resource
+        Override this method to implement custom handler selection logic, e.g. if multiple handlers were defined
+        for the same resource type.
```

Comment:
I just realized that iirc we want to drop this sometime soon (@wouterdb can you confirm?). Perhaps we should declare that it should generally not be overridden?
```suggestion
        Kept for backwards compatibility, new handler implementations should never override this.
```